### DISCUSSION TITLE
Maven plugin dying on me after SROAP04005

### DISCUSSION THREAD

**Discussion by cristalp** [smallrye/smallrye-open-api#1896] [Q&A] [answered]
I'm new to Smallrye, having used the Swagger annotations and a now dead Maven plugin. Well, "new" only for traditional JEE software, I've done a couple of Quarkus projects, and there everything works like a charm. I love Quarkus  :-)

Back to JEE: Migrating one REST service after another, from simple to complex, I'm stuck on Maven dying on me.

Simple REST services work well, but now it's more complex with several Maven modules containing other REST services or base classes for the module being built. This is the last thing I see, and then the build hangs:

```
Jul 03, 2024 5:40:09 PM io.smallrye.openapi.runtime.scanner.OpenApiDataObjectScanner process
WARN: SROAP04005: Could not find schema class in index: com.google.common.io.CharSource
```

I see this kind of message also when building simple modules. I'm not quite sure how to configure the plugin to ignore classes. `<scanExcludePackages>com.google.*</scanExcludePackages>` doesn't seem to do anything.

Maven debug output doesn't help any more. I see `Adding directories/artifacts to annotation scanner class loader:` and dependencies. But then the error message above appears, and that's it: The build hangs.

Can anyone help me?

**Comment by MikeEdgar** [comment]
Do you have a small project you can share that triggers the issue? I haven't seen/heard of it hanging like that before. Are you using version 3.11 of the plugin?

**Comment by cristalp** [comment]
Unfortunately, all the small projects work despite the warning. I am using the version 3.11.

Just guessing: If I could get rid of the warnings, perhaps the build would succeed?

**Reply by MikeEdgar** [reply]
I'm thinking the warning message is unrelated. It's usually harmless and would at worst result in the OpenAPI missing some object schema.

**Comment by cristalp** [comment]
I tried logging with Maven log level `-Dorg.slf4j.simpleLogger.defaultLogLevel=trace`, but I didn't see any more information.

A successfull build also shows the warnings, but then ends with
```
[INFO] Wrote the schema files to ....
```

So it probably dies in the scanning phase. Is there some way I can set the scanning to a more verbose logging?

**Comment by cristalp** [comment, marked as answer]
Well, now that I started migrating to SmallRye for real, the problem didn't occur anymore. I suspect there was a problem somewhere in the dependency chain where I had dependencies on the Swagger annotations.
